api/src/main/java/org/bahmni/module/hip/api/dao/EncounterDao.java

```diff
@@ -0,0 +1,12 @@
+package org.bahmni.module.hip.api.dao;
+
+import java.util.Date;
+import java.util.List;
+
+public interface EncounterDao {
+
+    List<Integer> GetEncounterIdsForVisitForPrescriptions(String patientUUID, String visit, Date fromDate, Date toDate) ;
```

Comment:
Is it possible to keep the GetEncounterIDs method generic so that the same can be use for other health-info ypes. Agreed the diagnostic report has a very specific encounter_type, but I believe that the other HI types can simply utilize the prescriptions SQL

Reply:
Since we don't know the tables for Discharge Summary and OP Consultation yet I don't think it is possible right now. We can always refactor when all HiTypes are done.